flatgfa-py/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+use flatgfa::flatgfa::{FlatGFA, GFABuilder, HeapStore};
+use pyo3::prelude::*;
+use pyo3::types::PyBytes;
+
+#[pyfunction]
+fn parse(filename: &str) -> PyFlatGFA {
+    let file = flatgfa::file::map_file(filename);
+    let store = flatgfa::parse::Parser::for_heap().parse_mem(file.as_ref());
+    PyFlatGFA(InternalStore::Heap(Box::new(store)))
+}
+
+#[pyfunction]
+fn load(filename: &str) -> PyFlatGFA {
+    let mmap = flatgfa::file::map_file(filename);
+    PyFlatGFA(InternalStore::File(mmap))
+}
+
+enum InternalStore {
+    Heap(Box<HeapStore>),
+    File(memmap::Mmap),
+}
+
+#[pyclass(frozen)]
+#[pyo3(name = "FlatGFA")]
+struct PyFlatGFA(InternalStore);
+
+#[pymethods]
+impl PyFlatGFA {
+    #[getter]
+    fn segments(self_: Py<Self>) -> SegmentList {
+        SegmentList { gfa: GFARef(self_) }
+    }
+}
+
+#[derive(Clone)]
+struct GFARef(Py<PyFlatGFA>);
+
+impl GFARef {
+    fn view(&self) -> FlatGFA {
+        // TK It seems wasteful to check the type of store every time... and to construct
+        // the view every time. It would be great if we could somehow construct the view
+        // once up front and hand it out to the various ancillary objects, but they need
+        // to be assured that the store will survive long enough.
+        match self.0.get().0 {
+            InternalStore::Heap(ref store) => store.view(),
+            InternalStore::File(ref mmap) => flatgfa::file::view(mmap),
+        }
+    }
+}
+
+#[pyclass]
+struct SegmentList {
+    gfa: GFARef,
+}
+
+#[pymethods]
+impl SegmentList {
```

Comment:
Out of curiosity: are these methods generated automatically or you wrote them down by hand? If its the latter, then how do you figure out if they're interacting safely with the Python API?